torch/distributed/rpc.py

```diff
@@ -0,0 +1,174 @@
+from . import invoke_rpc
+from . import ProcessGroupAgent
+
+import array
+import sys
+import torch
+
+
+_agent = None
+
+
+def _collect_worker_names(name, group):
+    from . import all_gather
+    from . import get_world_size
+
+    # collect name length
+    ws = get_world_size(group)
+    name_bytes = name if sys.version_info < (3, 0) else bytes(name, 'utf8')
+    name_bytes = list(array.array('B', name_bytes))
+    name_len = len(name_bytes)
+    len_input = torch.ones(1, dtype=torch.int64) * name_len
+    len_outputs = [torch.empty(1, dtype=torch.int64) for _ in range(ws)]
+    all_gather(len_outputs, len_input, group=group)
+
+    # collect name value
+    max_len = torch.stack(len_outputs).max().item()
+    name_input = torch.empty(max_len, dtype=torch.uint8)
+    name_input[:name_len] = torch.tensor(name_bytes, dtype=torch.uint8)
+    name_outputs = [torch.empty(max_len, dtype=torch.uint8) for _ in range(ws)]
+    all_gather(name_outputs, name_input, group=group)
+
+    names = []
+    for i in range(ws):
+        name_tensor = name_outputs[i][:len_outputs[i]]
+        names.append(bytearray(name_tensor.tolist()).decode('utf8'))
+
+    return names
+
+
+def join_rpc():
```

Comment:
As I mentioned above, this makes sense for ProcessGroup based RPCs, but not really something like a thrift implementation. We shouldn't have join and sync in the public APIs.

Reply:
I would assume Thrift also have a join or shutdown to make sure the local `RpcAgent` is not exiting too early?